src/native-common/Image.tsx

```diff
@@ -215,20 +215,20 @@ export class Image extends React.Component<Types.ImageProps, ImageState> impleme
             return this.props.source;
         }
 
-        const source: RN.ImageSourcePropType = { uri: this.props.source, headers: this.state.headers };
+        const source: RN.ImageSourcePropType = { uri: this.props.source, headers: this.state.headers || this.props.headers };
```

Comment:
Using `this.props.headers` if headers did not require customization. Saves us from cloning the incoming object.

Reply:
FWIW, there's no clone involved in the case where we use the value from this.props.headers, just a ref stuffed into state

Reply:
Yep, understood. What I meant is that we could clone `this.props.headers` and store the copy in state as an alternative way of fixing this.